kotlin/app/src/main/java/com/google/android/gms/samples/wallet/viewmodel/CheckoutViewModel.kt

```diff
@@ -1,63 +1,80 @@
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.google.android.gms.samples.wallet.viewmodel
 
 import android.app.Activity
 import android.app.Application
 import android.util.Log
 import androidx.lifecycle.AndroidViewModel
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
 import com.google.android.gms.common.api.ApiException
 import com.google.android.gms.pay.Pay
 import com.google.android.gms.pay.PayApiAvailabilityStatus
 import com.google.android.gms.pay.PayClient
 import com.google.android.gms.samples.wallet.util.PaymentsUtil
 import com.google.android.gms.tasks.Task
-import com.google.android.gms.wallet.*
+import com.google.android.gms.wallet.IsReadyToPayRequest
+import com.google.android.gms.wallet.PaymentData
+import com.google.android.gms.wallet.PaymentDataRequest
+import com.google.android.gms.wallet.PaymentsClient
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 
 class CheckoutViewModel(application: Application) : AndroidViewModel(application) {
 
+    data class State(
+        val googlePayAvailable: Boolean? = false,
+        val googleWalletAvailable: Boolean? = false,
+        val googlePayButtonClickable: Boolean = true,
+        val googleWalletButtonClickable: Boolean = true,
+        val checkoutSuccess: Boolean = false,
+    )
+
+    private val _state = MutableStateFlow(State())
```

Comment:
Can the underscore be avoided? It's used in some props below because they have counterparts with duplicate names, but generally it's not a convention on Java or Kotlin, since there's privacy modifiers.